### ISSUE TITLE
Handling bot user when handling github webhook

### ISSUE THREAD

**Issue by begedin** [code-corps/code-corps-api#939]
# Problem

If a CodeCorps user who is not connected to github creates a task on codecorps which is connected to github this triggers the following process.

1. Task is created
2. Github issue is created on selected repo, authored by the github application bot user
3. Github webhook for `issue_created` is triggered and handled from our end
4. When handling the webhook, the bot user record is created on codecorps, since there's no matching actual user.

Step 4 should not happen, or should be reconsidered, at least.

The problem is in `github/event/issues.ex`

The current implementation is 

```Elixir
Multi.new
|> Multi.run(:repo, fn _ -> RepoFinder.find_repo(payload) end)
|> Multi.run(:user, fn _ -> UserLinker.find_or_create_user(payload) end)
|> Multi.run(:tasks, fn %{repo: github_repo, user: user} -> TaskSyncer.sync_all(github_repo, user, payload) end)
```

This works then the task creator is not the bot. It however, does not work when it is the bot, since in that case, we will already have a matching task record which is associated to an actual, github unconnected user.

## Scenarios

The most complicated scenario is

- `org_1/project` and `org_2/project` are both connected to `repo_1`
- `user_1` is member of `org_1`, but not `org_2`, they are not github connected
. `user_1` creates `task_1` on `org_1/project`
- this causes `issue_1` to be created on `repo_1`
- this causes an `issues` webhook to be fired, by `app[bot] user`
- we handle the webhook
    - `org_1/project/task_1` already exist, so we should find it and "update" it, the user should remain `user_1`
    - `org_2/project/task_1` does not exist yet, so it should be created. `user_1` is not a member of `org_2`, so the user assigned to the task should be the bot user? (OPTION A) Or should we still assign `user_1` and allow him to edit that specific task on code corps to, even though they aren't a member of `org_2`? (OPTION B)

Other scenarios are 

- there's just one project linked to the repo (easy, find and update the task)
- there are multiple projects linked, but the user is a member of all organizations (find and update initial task, create other ones, associate with user)

#### if issue user is the bot and we went with option A

- get task identifier (`github_repo_name`, `github_repo_owner`, `github_issue_number`)
- find task for that identifier, there needs to be one, "update" it (probably no changes)
- for each other project linked to that repo, create or update task and assign to project, repo and the bot user

#### if issue user is the bot and we went with option B

- get task identifier (`github_repo_name`, `github_repo_owner`, `github_issue_number`)
- find task for that identifier, there needs to be one, "update" it (probably no changes)
- get user for that task
- for each other project linked to that repo, create or update task and assign to project, repo and the initial task user
 
### if issue user is not the bot

Basically, use the current flow

- get repo
- find or create user
- for each project associated with repo
  - create task, associate with project, repo and user

### Notes 

@JoshSmith  I'm marking this as "question", since it probably needs further discussion or at least a review of the process I'm proposing.

**Comment by joshsmith** [comment]
> Or should we still assign `user_1` and allow him to edit that specific task on code corps too, even though they aren't a member of `org_2`? (OPTION B)

I think this makes the most sense. I don't think our bot user should ever be assigned to tasks on Code Corps. That indicates something seriously wrong happened.

The question here is how complex does this make our policies?

**Comment by joshsmith** [comment]
The answer to my last question is: not at all. `task |> task_authored_by?(user)` is the logic for `update?`. Anyone can create tasks on projects, just like anyone can create issues on GitHub.

**Comment by joshsmith** [comment]
Given what's happening in #925 I'm concerned that the task's user is getting overwritten in what's already in `develop`. I'm not sure I picked up on it in my testing.

Looking at it now, it does not _appear_ to overwrite the user, so I'm wondering what's different in our task update code from GitHub webhooks vs our comment update code.
